tox.ini

```diff
@@ -4,23 +4,39 @@
 [tox]
 isolated_build = True
 skip_missing_interpreters = True
-envlist = build-json-schemas, unit, lint
+envlist = build-json-schemas, unit, lint, integration
 
 
 [vars]
 tst_path = {toxinidir}/tests
 docs_path = {toxinidir}/docs
-all_path = {[vars]tst_path} {[vars]docs_path}
+tester_path = {toxinidir}/tester
+all_path = {[vars]tst_path} {[vars]docs_path} {toxinidir}/run_matrix.py
 
 [testenv:unit]
 description = run unittests
 deps =
-    pytest
+    .[unit_tests]
+    # todo replace with pypi package when available
     git+https://github.com/canonical/interface-tester-pytest
 setenv =
     PYTHONPATH={toxinidir}
 commands =
-    pytest {[vars]tst_path}
+    pytest {[vars]tst_path}/test_unit
+
+
+[testenv:integration]
```

Comment:
Where will integration tests run? In the CI? `tox -e integration` crashes when I run it locally with the following error:

```
ERROR  - ModuleNotFoundError: No module named 'charm'
```

Reply:
Can you show the full output of `tox -re integration` please?

Reply:
```
(venv) guillaume@thinkpad:~/PycharmProjects/charm-relation-interfaces-pietro$ tox -re integration
integration recreate: /home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/integration
integration installdeps: .[unit_tests], git+https://github.com/canonical/interface-tester-pytest
integration installed: charm-relation-interfaces @ file:///home/guillaume/PycharmProjects/charm-relation-interfaces-pietro,click==8.1.3,iniconfig==2.0.0,ops==2.2.0,ops-scenario==2.2,packaging==23.1,pluggy==1.0.0,pydantic==1.10.7,pytest==7.3.0,pytest-interface-tester @ git+https://github.com/canonical/interface-tester-pytest@1798fad829155bc9a97a324e80bba2c40434acec,PyYAML==6.0,typer==0.7.0,typing_extensions==4.5.0,websocket-client==1.5.1
integration run-test-pre: PYTHONHASHSEED='2394826534'
integration run-test: commands[0] | pytest ./tests
============================================================================================================== test session starts ==============================================================================================================
platform linux -- Python 3.11.2, pytest-7.3.0, pluggy-1.0.0
cachedir: .tox/integration/.pytest_cache
rootdir: /home/guillaume/PycharmProjects/charm-relation-interfaces-pietro
plugins: interface-tester-0.1
collected 0 items / 1 error                                                                                                                                                                                                                     

==================================================================================================================== ERRORS =====================================================================================================================
_________________________________________________________________________________________________________ ERROR collecting test session _________________________________________________________________________________________________________
/usr/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1206: in _gcd_import
    ???
<frozen importlib._bootstrap>:1178: in _find_and_load
    ???
<frozen importlib._bootstrap>:1149: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:690: in _load_unlocked
    ???
.tox/integration/lib/python3.11/site-packages/_pytest/assertion/rewrite.py:172: in exec_module
    exec(co, module.__dict__)
tests/test_integration/mock_charm/tests/test_interfaces/conftest.py:5: in <module>
    from charm import MyCharm
E   ModuleNotFoundError: No module named 'charm'
============================================================================================================ short test summary info ============================================================================================================
ERROR  - ModuleNotFoundError: No module named 'charm'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
=============================================================================================================== 1 error in 0.07s ================================================================================================================
ERROR: InvocationError for command /home/guillaume/PycharmProjects/charm-relation-interfaces-pietro/.tox/integration/bin/pytest ./tests (exited with code 2)
____________________________________________________________________________________________________________________ summary ____________________________________________________________________________________________________________________
ERROR:   integration: commands failed
```

Reply:
Those were python path related issues, I just added the suggestions/fixes in the tox.ini file

Reply:
The other comment I had related to this was "where/how" will those integration tests run? As far as I saw, those tests aren't referenced in the CI or in documentation

Reply:
Added them in the quality checks pipeline